src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1338,9 +1464,13 @@ static void processContextRegistrationElement (BSONObj cr, EntityIdVector enV, A
 
     /* Note that attributes can be included only if at least one entity has been found */
     if (crr.contextRegistration.entityIdVector.size() > 0) {
-        std::vector<BSONElement> queryAttrV = cr.getField(REG_ATTRS).Array();
-        for (unsigned int ix = 0; ix < queryAttrV.size(); ++ix) {
+        if (cr.hasField(REG_ATTRS)) /* To prevent registration in the E-<null> style */
+        {
+          std::vector<BSONElement> queryAttrV = cr.getField(REG_ATTRS).Array();
+          for (unsigned int ix = 0; ix < queryAttrV.size(); ++ix)
+          {
             processAttribute(&crr, attrL, queryAttrV[ix].embeddedObject());
+          }
```

Comment:
The new line is this:

cr.hasField(REG_ATTRS)

Is not cheking cero-sized... is checking that the field REG_ATTRS exists in the cr BSONObj. We need that, because if the field doesn't exists... segfault.